rust/src/patches.rs

```diff
@@ -34,14 +34,24 @@ pub enum PatchAction {
         obj: ObjId,
         index: u64,
         values: Vec<Value>,
-        marks: HashMap<String, Value>,
     },
     SpliceText {
         obj: ObjId,
         index: u64,
         value: String,
         marks: HashMap<String, Value>,
     },
+    JoinBlock {
+        index: u64,
+        // ? cursor
```

Comment:
@alexjg I got far enough to get things compiling, but not far enough to know how to reasonably translate these new Patch types out into Swift (or even much of a sense how they might be used). Struggled with what Cursor represents independently from index (and how to present it), as well as the Patch embedded within UpdateBlock.